update-vo-config/update-vo-config

```diff
@@ -0,0 +1,507 @@
+        content += format("    ),\n")
+    return content
+
+
+def write_vo_template(vo_details, basedir, namespaces):
+    """Write VO template
+    """
+    vo_params_ns = namespaces['params_dir'] + os.path.sep + vo_details['name']
+    vo_params_tpl = basedir + os.path.sep + vo_params_ns + ".pan"
+    force_voms_admin = False
+
+    sys.stdout.write("Writing template for VO " + vo_details['name'] +
+                     " (" + vo_params_tpl + ")\n")
+    template = open(vo_params_tpl, 'w')
+    template.write("structure template " + vo_params_ns + ";\n\n")
+    template.write("'name' ?= '" + vo_details['name'] + "';\n")
+    vo_name = re.sub(re.compile('^vo\.'), '', vo_details['name'])
+    account_prefix = vo_name[0:3] + to_base26(vo_details['id'])
+    template.write("'account_prefix' ?= '" + account_prefix + "';\n\n")
+
+    # VOMS Servers
+    template.write("'voms_servers' ?= list(\n")
+    if not vo_details['voms_endpoints']:
+        sys.stderr.write("WARNING: VO " + vo_details['name'] +
+                         " has no VOMS endpoint defined\n")
+    elif len(vo_details['voms_endpoints']) == 1:
+        force_voms_admin = True
+    for endpoint in vo_details['voms_endpoints']:
+        template.write("    dict(\n")
+        template.write("        'name', '%s',\n" % endpoint.hostname)
+        template.write("        'host', '%s',\n" % endpoint.hostname)
+        template.write("        'port', %s,\n" % endpoint.vomses_port)
+        template.write("        'adminport', %s,\n" % endpoint.https_port)
+        if not endpoint.is_admin:
+            if not force_voms_admin:
+                template.write("        'type', list('voms-only'),\n")
+            else:
+                sys.stdout.write("WARNING: voms-admin enabled on " +
+                                 "%s as this is the " % endpoint.hostname +
+                                 "only VOMS server (VO card " +
+                                 "inconsistency)\n")
+        template.write("    ),\n")
+    template.write(");\n\n")
+
+    # VOMS Mapping
+    template.write("'voms_mappings' ?= list(\n")
+    legacy_suffix = []
+    for mapping in vo_details['voms_mapping']:
+        suffix = get_legacy_suffix(vo_details['id'], mapping, legacy_suffix)
+        legacy_suffix.append(suffix)
+        suffix2 = get_account_suffix(vo_details['name'], mapping)
+        template.write("    dict(\n")
+        template.write("        'description', '%s',\n" % mapping['desc'])
+        template.write("        'fqan', '%s',\n" % mapping['expr'])
+        template.write("        'suffix', '%s',\n" % suffix)
+        template.write("        'suffix2', '%s',\n" % suffix2)
+        template.write("    ),\n")
+
+    template.write(");\n\n")
+
+    # Base UID
+    base_uid = vo_details['id'] * 1000
+    template.write("'base_uid' ?= %i;\n" % base_uid)
+    template.close()
+
+
+def update_voms_servers(vo_details, basedir, namespaces, debug=False):
```

Comment:
This function is quite large and complex, could it be split up into smaller sub-functions?

Reply:
The function has been split in commit 9ac2376334a3ed9bde76d6ae050aedc9f4c8d063.